pkg/netlink/netlink_watcher.go

```diff
@@ -0,0 +1,2053 @@
+			if bp != "" {
+				BP, _ = infradb.GetBP(bp)
+			}
+		}
+		if BP != nil {
+			neighbor.Type = SVI
+			neighbor.Metadata["vport_id"] = BP.Metadata.VPort
+			neighbor.Metadata["vlanID"] = uint32(vlanID)
+			neighbor.Metadata["portType"] = BP.Spec.Ptype
+		} else {
+			neighbor.Type = IGNORE
+		}
+	} else if strings.HasPrefix(NameIndex[neighbor.Neigh0.LinkIndex], path.Base(neighbor.VrfName)) && neighbor.Protocol == zebraStr {
+		pattern := fmt.Sprintf(`%s-\d+$`, path.Base(neighbor.VrfName))
+		mustcompile := regexp.MustCompile(pattern)
+		s := mustcompile.FindStringSubmatch(neighbor.Dev)
+		var LB *infradb.LogicalBridge
+		vID := strings.Split(s[0], "-")[1]
+		lbs, _ := infradb.GetAllLBs()
+		vlanID, err := strconv.ParseUint(vID, 10, 32)
+		if err != nil {
+			panic(err)
+		}
+		for _, lb := range lbs {
+			if lb.Spec.VlanID == uint32(vlanID) {
+				LB = lb
+				break
+			}
+		}
+		if LB.Spec.Vni != nil {
+			vid, err := strconv.Atoi(vID)
+			if err != nil {
+				panic(err)
+			}
+			fdbEntry := latestFDB[fdbKey{vid, neighbor.Neigh0.HardwareAddr.String()}]
+			neighbor.Metadata["l2_nh"] = fdbEntry.Nexthop
+			neighbor.Type = VXLAN // confirm this later
+		}
+	} else if path.Base(neighbor.VrfName) == "GRD" && phyFlag && neighbor.Protocol != zebraStr {
+		VRF, _ := infradb.GetVrf("//network.opiproject.org/vrfs/GRD")
+		r, ok := lookupRoute(neighbor.Neigh0.IP, VRF)
+		if ok {
+			if r.Nexthops[0].nexthop.LinkIndex == neighbor.Neigh0.LinkIndex {
+				neighbor.Type = PHY
+				neighbor.Metadata["vport_id"] = phyPorts[NameIndex[neighbor.Neigh0.LinkIndex]]
+			} else {
+				neighbor.Type = IGNORE
+			}
+		} else {
+			neighbor.Type = OTHER
+		}
+	}
+	return neighbor
+}
+
+// parseNeigh parses the neigh
+func parseNeigh(nm []neighIPStruct, v string) neighList {
+	var NL neighList
+	for _, ND := range nm {
+		var ns neighStruct
+		ns.Neigh0.Type = OTHER
+		ns.VrfName = v
+		if ND.Dev != "" {
+			vrf, _ := vn.LinkByName(ND.Dev)
+			ns.Neigh0.LinkIndex = vrf.Attrs().Index
+		}
+		if ND.Dst != "" {
+			ipnet := &net.IPNet{
+				IP: net.ParseIP(ND.Dst),
+			}
+			ns.Neigh0.IP = ipnet.IP
+		}
+		if len(ND.State) != 0 {
+			ns.Neigh0.State = getState(ND.State[0])
+		}
+		if ND.Lladdr != "" {
+			ns.Neigh0.HardwareAddr, _ = net.ParseMAC(ND.Lladdr)
+		}
+		if ND.Protocol != "" {
+			ns.Protocol = ND.Protocol
+		}
+		//	ns  =  neighborAnnotate(ns)   /* Need InfraDB to finish for fetching LB/BP information */
+		ns.Key = neighKey{VrfName: v, Dst: ns.Neigh0.IP.String(), Dev: ns.Neigh0.LinkIndex}
+		if preFilterNeighbor(ns) {
+			NL.NS = append(NL.NS, ns)
+		}
+	}
+	return NL
+}
+
+func notifyAddDel(r interface{}, event string) {
+	log.Printf("netlink: Notify event: %s\n", event)
+	EventBus.Publish(event, r)
+}
+
+func deepCheck(v1 interface{}, v2 interface{}) bool {
+	if reflect.TypeOf(v1) != reflect.TypeOf(v2) {
+		return false
+	}
+	switch t := v1.(type) {
+	case *RouteStruct:
+		return t.deepEqual(v2.(*RouteStruct), true)
```

Comment:
just use t here

Reply:
can you please detail on how and where to use t ?

Reply:
my oversight, please ignore this comment